builder/confd/confd.go

```diff
@@ -0,0 +1,91 @@
+// Package confd provides basic Confd support.
+//
+// Right now, this library is highly specific to the needs of the present
+// builder. Because the confd library is not all public, we don't use it directly.
+// Instead, we invoke the CLI.
+package confd
+
+import (
+	"github.com/Masterminds/cookoo"
+	"github.com/Masterminds/cookoo/log"
+	"github.com/Masterminds/cookoo/safely"
+
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+)
+
+// RunOnce runs the equivalent of `confd --onetime`.
+//
+// This may run the process repeatedly until either we time out (~20 minutes) or
+// the templates are successfully built.
+//
+// Importantly, this blocks until the run is complete.
+//
+// Params:
+// - node (string): The etcd node to use. (Only etcd is currently supported)
+//
+// Returns:
+// - The []bytes from stdout and stderr when running the program.
+//
+func RunOnce(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+
+	node := p.Get("node", "127.0.0.1:4001").(string)
+
+	dargs := []string{"-onetime", "-node", node, "-log-level", "error"}
+
+	fmt.Println("Building confd templates. This may take a moment.")
+
+	limit := 1200
+	timeout := time.Second * 3
+	var lasterr error
+	for i := 0; i < limit; i++ {
+		if out, err := exec.Command("confd", dargs...).CombinedOutput(); err == nil {
+			log.Infof(c, "Templates generated for %s on run %d", node, i)
+			return out, nil
+		} else {
+			log.Debugf(c, "Recoverable error: %s", err)
+			log.Debugf(c, "Output: %q", out)
+			lasterr = err
+		}
+
+		time.Sleep(timeout)
+		log.Infof(c, "Re-trying template build. (Elapsed time: %d)", i*3)
```

Comment:
That was just for debugging, but I will fix it and leave it in there. It might be useful to know.